src/installer/records.py

```diff
@@ -175,13 +175,15 @@ def from_elements(cls, path: str, hash_: str, size: str) -> "RecordEntry":
         if not path:
             issues.append("`path` cannot be empty")
 
+        hash_value: Optional[Hash] = None
         if hash_:
             try:
-                hash_value: Optional[Hash] = Hash.parse(hash_)
+                hash_value = Hash.parse(hash_)
+                if hash_value.name not in hashlib.algorithms_available:
+                    issues.append(f"invalid hash algorithm '{hash_value.name}'")
+                    hash_value = None
```

Comment:
Should this be performing this validation in a validate command, since this would prevent a `RecordEntry` from being created with a hash that isn't in `algorithms_available`?

Reply:
My thinking was that it's desirable to be unable to create a `RecordEntry` with a bad hash algorithm, but I'm happy to be guided in another direction if you have a preference.

Yet another approach, doubling down on "invalid hash algorithms are invalid" would be to raise a `ValueError` from `Hash.parse()`, though the error message from that path isn't currently very informative